### DISCUSSION TITLE
Limitations for development

### DISCUSSION THREAD

**Discussion by DjordjeMandic** [Ralim/IronOS#1018] [Q&A] [answered]
Alright i would like to start a discussion where we would discuss what problems are we currently facing in development of IronOS. 

For example flash limitations, software bugs and etc....

I know there is flash limitation on MiniWare soldering irons but im thinking maybe we can chose different MCU that is pin compatible with more features and etc...

**Comment by Ralim** [comment, marked as answer]
There isn't any really big development stoppers, there is still some space left in all devices.

Generally most users will not swap out the mcu.

On the miniware units there is the possibility of going beyond the 64k, as STM _usually_ makes the parts with 128k, but then downgrades units with flash errors to 64k. Most of the time you can use the extra flash _okay_ but you roll a dice on it randomly failing / having stuck bits etc. I think you would also need to replace the stock bootloader.
